src/com/sap/piper/ConfigurationMerger.groovy

```diff
@@ -12,6 +14,22 @@ class ConfigurationMerger {
         return merged
     }
 
+    @NonCPS
+    def static merge(Map configs, Map configKeys, Map defaults = [:]) {
```

Comment:
Why is configKeys a map? It is IMO a Set.

&lt;by_the_way&gt;in the other merge (Map, List, Map) it is also not good to have a list for the config keys. Should be a set, since it does never make sense to have duplicate config keys, and the order of the entries is not significant. But this is another story.&lt;/by_the_way&gt;

Having it as a List we could re-use the other merge method. I guess filtering out null values is fine also here.

Reply:
How could I describe a deep structure with keys and its child keys in a list?

Reply:
I'm talking *only* about the keys. A feature of the config merging is dropping of keys which are present in the configuration, but not required by the step. And these keys should be described in a set, not in a list, since there are no duplicates and the order is not significant.

E.g. at the moment there are the configKeys here designed as a Map, but only the key set of this map is used. That is weird.

In case we would like to filter also in  a deep-merge it sounds reasonable to have a Map. In this case we specify the subkeys which are expected on the next level. This is not fully un-reasonable ...

Reply:
keymap values are used here: https://github.com/SAP/jenkins-library/pull/66/files#diff-07e7b5ab7692947662654f171daa514fR24

Reply:
Understood, when we introduce deep merge it is a map. The keys, which we have now in a list are not contained in a set (.. this was basially my point), the value associated with the entry in the map is in turn a map for the next level of merging.